__tests__/Unit/hooks/tasksApi.test.tsx

```diff
@@ -357,6 +359,70 @@ describe('useUpdateTaskMutation()', () => {
     }, 7000);
 });
 
+describe('useUpdateSelfTaskMutation', () => {
+    const updatedSelfTaskData = {
+        status: 'COMPLETED',
+    };
+    test('updates a task', async () => {
+        const taskId = '1eJhUW19D556AhPEpdRd';
+        const { result, waitForNextUpdate } = renderHook(
+            () => useUpdateSelfTaskMutation(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const [updateTask, initialResponse] = result.current;
+        expect(initialResponse.isLoading).toBe(false);
+        expect(initialResponse.data).toBeUndefined();
+
+        act(() => {
+            const task = {
+                task: updatedSelfTaskData,
+                id: taskId,
+            };
+            updateTask(task);
+        });
+
+        const loadingResponse = result.current[1];
+        expect(loadingResponse.isLoading).toBe(true);
+
+        await waitForNextUpdate({ timeout: 7000 });
+
+        const nextResponse = result.current[1];
+        expect(nextResponse.data).toBeNull();
+        expect(nextResponse.isSuccess).toBe(true);
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.status).toBe(QueryStatus.fulfilled);
+    }, 7000);
```

Comment:
can we check for status 204 here?

Reply:
even status of 200 will give a fulfilled status we need to

Reply:
but with the handler returning 204 on success then why should we consider for 200 status code?